### ISSUE TITLE
Unable to save edited product when max_sale_qty is Magento's default

### ISSUE THREAD

**Issue by sol0mandra** [magento/magento2#23319]
### Preconditions (*)

1. Install Magento Open Source 2.3.1 

### Steps to reproduce (*)

1. Go to admin area
2. Go to Stores->Configuration->Catalog->Inventory->Product Stock Options
3. Edit the global default to be 0 
4. Save the config
5. Open create a simple product form and fill in minimum required data
6. Open the advanced inventory popup
7. Uncheck the checkbox "use_config_max_sale_qty"
8. Observe the default value of 0 in max_sale_qty field
9. Do not edit the default provided
10. Click Save

### Expected result (*)

1. Product is successfully saved

### Actual result (*)

1. Save is not happening, validation is failed due to validate-grater-then-0 on max_sale_qty field.

This value however comes from the cataloginventory's db_schema (module-catalog-inventory/etc/db_schema.xml):
```
<table name="cataloginventory_stock_item" resource="default" engine="innodb" comment="Cataloginventory Stock Item">
...
<column xsi:type="decimal" name="max_sale_qty" scale="4" precision="12" unsigned="false" **nullable="false"
                default="0"** comment="Max Sale Qty"/>
...
</table>
```

While in the same module, in magento/module-catalog-inventory/view/adminhtml/ui_component/product_form.xml:
```
<field name="max_sale_qty" formElement="input">
    <argument name="data" xsi:type="array">
        <item name="config" xsi:type="array">
            <item name="value" xsi:type="object">Magento\CatalogInventory\Model\Source\StockConfiguration</item>
        </item>
    </argument>
    <settings>
        <scopeLabel>[GLOBAL]</scopeLabel>
        <validation>
            <rule name="**validate-greater-than-zero**" xsi:type="boolean">true</rule>
        </validation>
        <label translate="true">Maximum Qty Allowed in Shopping Cart</label>
        <dataScope>max_sale_qty</dataScope>
    </settings>
</field>
```

And in the same module, inmodule-catalog-inventory/etc/adminhtml/system.xml:
```
<section id="cataloginventory" translate="label" type="text" sortOrder="50" showInDefault="1" showInWebsite="1" showInStore="1">
...
<group id="item_options" translate="label comment" type="text" sortOrder="10" showInDefault="1" showInWebsite="1" showInStore="1">
...
<field id="max_sale_qty" translate="label" type="text" sortOrder="4" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
                    <label>Maximum Qty Allowed in Shopping Cart</label>
                    <validate>**validate-number**</validate>
                </field>


So, there is inconsistency - if 0 is allowed as value for max_sale_qty. If 0 is set from the start as default, there shouldn't be validation that would forbid it on the frontend.

**Comment by sol0mandra** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @eshwilk. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @eshwilk, here is your [Magento instance](https://i-23319-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23319-2-3-develop.instances.magento-community.engineering/admin](https://i-23319-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sol0mandra** [comment]
The problem manifests itself and is even more important if that's a project, where products are created programatically (like a lot of migration projects). Then the DB default of 0 for max_sale_qty is set for all freshly created products - and in this way breaking any editing of the products later on

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17605`, `MC-17606` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sol0mandra** [comment]
The problem seems to be introduced with this fix: https://github.com/magento/magento2/issues/18477

**Comment by magento-engcom-team** [comment]
Hi @eshwilk. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.2-develop` branch(es).
Related commit(s):
 - [a774aad16f4a26fe6fa535abf70af948634195be](https://github.com/magento/magento2/commit/a774aad16f4a26fe6fa535abf70af948634195be)
 - [0ad43ac5682d94f791c38ebaa590eb7f7372f432](https://github.com/magento/magento2/commit/0ad43ac5682d94f791c38ebaa590eb7f7372f432)
 - [6db5533b5708e0c9db20b5a6db4fa98b42d80d78](https://github.com/magento/magento2/commit/6db5533b5708e0c9db20b5a6db4fa98b42d80d78)

**Comment by magento-engcom-team** [comment]
Hi @eshwilk. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.3-develop` branch(es).
Related commit(s):
 - [778b42b6b9ce7f083c7e7adf45eb588519b7e538](https://github.com/magento/magento2/commit/778b42b6b9ce7f083c7e7adf45eb588519b7e538)
 - [931af1d74c659912c5bad7477ca7070fb1488f60](https://github.com/magento/magento2/commit/931af1d74c659912c5bad7477ca7070fb1488f60)
 - [eeb60b0ec5b5dc9df7b2f2e03adffefa1dc37a8a](https://github.com/magento/magento2/commit/eeb60b0ec5b5dc9df7b2f2e03adffefa1dc37a8a)
 - [2ac7ada06a077c7d47cb38b035ef40a8a75402f0](https://github.com/magento/magento2/commit/2ac7ada06a077c7d47cb38b035ef40a8a75402f0)
 - [a15f822811cf19b7673dc05ac7c110380730cc58](https://github.com/magento/magento2/commit/a15f822811cf19b7673dc05ac7c110380730cc58)
 - [24ae0a99a302105bb913fed3b0d8b52a23cae780](https://github.com/magento/magento2/commit/24ae0a99a302105bb913fed3b0d8b52a23cae780)
 - [75dce6df5072ccf54ea8c65d6f34af8b7b2dc562](https://github.com/magento/magento2/commit/75dce6df5072ccf54ea8c65d6f34af8b7b2dc562)
 - [807cb5cc9babed76a93f99b5b917d8db03939adb](https://github.com/magento/magento2/commit/807cb5cc9babed76a93f99b5b917d8db03939adb)

The fix will be available with the upcoming `2.3.4` release.
